server/src/main/java/org/elasticsearch/plugins/PluginsService.java

```diff
@@ -281,35 +304,75 @@ public int hashCode() {
     }
 
     /**
-     * Extracts all installed plugin directories from the provided {@code rootPath} expanding meta plugins if needed.
+     * Represents a meta-plugin and the {@link Bundle}s corresponding to its constituents.
+     */
+    static class MetaBundle implements BundleCollection{
```

Comment:
nit: missing space after BundleCollection

Reply:
Thanks @jimczi I will correct this.

Reply:
I pushed 368f5f7d8ff1885b33453da9f64900847c01d6ce.